### PR TITLE
openldap: fix cross-compilation

### PR DESCRIPTION
**Pull request by Mindavi**
###### Description of changes

Fixes cross-compilation for openldap. Openldap is also a dependency of udisks2.

Staging-next: https://github.com/NixOS/nixpkgs/pull/173605/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux (cross-compiled)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/openldap/default.nix [hunk 1]
```diff
@@ -61,7 +61,8 @@ stdenv.mkDerivation rec {
 
   makeFlags= [
     "CC=${stdenv.cc.targetPrefix}cc"
-    "STRIP="  # Disable install stripping as it breaks cross-compiling. We strip binaries anyway in fixupPhase.
+    "STRIP="       # Disable install stripping as it breaks cross-compiling. We strip binaries anyway in fixupPhase.
+    "STRIP_OPTS="  # Disable install stripping as it breaks cross-compiling. We strip binaries anyway in fixupPhase.
```
Comment (mweinelt) [inline pkgs/development/libraries/openldap/default.nix:65]:
```suggestion
    # Disable install stripping as it breaks cross-compiling. We strip binaries anyway in fixupPhase.
    "STRIP="       
    "STRIP_OPTS="
```

Commenting it once is probably sufficient.

**Comment by vcunat** [comment]
Perhaps conditionally on cross only?  I'm not too fond of the rebuild amount at this stage, if it improves basically nothing (on those rebuild platforms).

File: pkgs/development/libraries/openldap/default.nix (hunk 1 above)
Comment (Mindavi) [reply pkgs/development/libraries/openldap/default.nix]: Done

**Comment by Mindavi** [comment]
Thanks. I agree, made it conditional for now.

**Comment by dasJ** [comment]
I'm a bit afraid to last-minute merge this, altough the breakages should be minimal.
Maybe we should wait until the branch-off (this evening european time), target staging with this and assign the `backport staging-22.05` label?

**Comment by Mindavi** [comment]
This should have the same effect as the previously added `STRIP=` in makeFlags, as far as I can tell (didn't research in any way), the flag was just renamed from `STRIP` to `STRIP_OPTS`.

Since it's now conditional on cross-compiling, it shouldn't affect _any_ native packages.

I don't really mind if we do this now or after the release (possibly with a backport). If we do it after the release I will make it unconditional again. I'm not in a rush. But I saw it broke on staging-next where master was fine, and I'm trying to keep cross regressions low.

**Comment by vcunat** [comment]
Yes, the effect is cross-only (including rebuilds), and even there it doesn't seem risky.

**Comment by dasJ** [comment]
After talking to @mweinelt, it seems to be a good idea to target `master` with this PR and to merge it once staging-next is in, backporting this to release-22.05.
This way we can skip staging altogether because it's not a mass-rebuild anymore.

**Comment by mweinelt** [comment]
I don't mind either as it doesn't cause rebuilds on hydra anyway.

**Comment by vcunat** [comment]
It's only targeted at `staging-next` because it's only needed after the update that's in `master..staging-next` now.

**Comment by dasJ** [comment]
Alright, then

**Pull request merged**